gst/tensor_filter/tensor_filter.c

```diff
@@ -354,44 +352,37 @@ gst_tensor_filter_fix_caps (GstTensor_Filter * filter, gboolean isInput,
   staticcap = gst_static_caps_get (&rawcap);
 
   if (isInput == TRUE) {
-    type = prop->inputMeta.types;
-    int i;
-    for (i = 0; i < NNS_TENSOR_SIZE_LIMIT; i++) {
-      dimension[i] = prop->inputMeta.dims[i];
-    }
+    meta = &prop->inputMeta;
     configured = prop->inputConfigured & _TFC_ALL;
   } else {
-    type = prop->outputMeta.types;
-    int i;
-    for (i = 0; i < NNS_TENSOR_SIZE_LIMIT; i++) {
-      dimension[i] = prop->outputMeta.dims[i];
-    }
+    meta = &prop->outputMeta;
     configured = prop->outputConfigured & _TFC_ALL;
   }
 
   /* 2. configure caps based on type & dimension */
   if (configured == _TFC_ALL) {
-    rank = gst_tensor_filter_get_rank (dimension[0]);
+    rank = gst_tensor_filter_get_rank (meta->dims[0]);
```

Comment:
You didn't change it with this PR; but this is suspicious.

Is it ok to use the first tensor only?
(not using other ```dimension[i]``` for fixing the cap)

Reply:
Not actually, it has to be changed to care multiple tensors. I'm planning to change the whole element and it just preparation.